content/sklearn/concepts/probability-calibration/probability-calibration.md

```diff
@@ -0,0 +1,146 @@
+---
+Title: 'Probability Calibration'
+Description: 'Probability calibration improves the reliability of predicted probabilities from machine learning models.'
+Subjects:
+  - 'Data Science'
+  - 'Machine Learning'
+Tags:
+  - 'Machine Learning'
+  - 'Scikit-learn'
+  - 'Supervised Learning'
+  - 'Unsupervised Learning'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+In Sklearn, **Probability Calibration** is a technique used to improve the reliability of predicted probabilities from machine learning models. When a model outputs a probability, it's essentially making a statement about the likelihood of a specific outcome. A well-calibrated model ensures that these probabilities accurately reflect the true likelihoods.
+
+Sklearn provides two primary methods for implementing probability calibration:
+
+- **Platt Scaling**: Fits a logistic regression model to the model's output probabilities.
+- **Isotonic Regression**: Fits a non-parametric isotonic regression model to the model's output probabilities.
+
+## Syntax
+
+The `CalibratedClassifierCV` class is used to implement probability calibration.
+
+Following is the syntax for implementing probability calibration using Platt Scaling:
+
+```pseudo
+from sklearn.calibration import CalibratedClassifierCV
+from sklearn.linear_model import LogisticRegression
+
+# Create a logistic regression classifier
+model = LogisticRegression()
+
+# Calibrate the classifier using Platt Scaling
+model_calibrated = CalibratedClassifierCV(model, cv=5, method="sigmoid")
+
+# Fit the calibrated classifier to the training data
+model_calibrated.fit(X_train, y_train)
+
+# Make predictions using the calibrated classifier
+y_pred_prob = model_calibrated.predict_proba(X_test)
+```
+
+Following is the syntax for implementing probability calibration using Isotonic Regression:
```

Comment:
We should have more information about how the method parameter relates here and what is Isotonic regression.

```
Isotonic regression is a non-parametric regression technique that fits a piecewise constant, monotonic (increasing or decreasing) function to the data. 

In the context of calibration, the isotonic method uses isotonic regression to map the model's raw probabilities to calibrated probabilities while preserving their relative order.
```